### PR TITLE
Fix column names mismatch in biplot_scores

### PR DESCRIPTION
**Pull request by essicolo**
Following issue [1519](https://github.com/biocore/scikit-bio/issues/1519).

Please complete the following checklist:

* [x] I have read the guidelines in [CONTRIBUTING.md](https://github.com/biocore/scikit-bio/blob/master/CONTRIBUTING.md).

* [x] I have documented all public-facing changes in [CHANGELOG.md](https://github.com/biocore/scikit-bio/blob/master/CHANGELOG.md).

* [ ] **This pull request includes code, documentation, or other content derived from external source(s).** If this is the case, ensure the external source's license is compatible with scikit-bio's [license](https://github.com/biocore/scikit-bio/blob/master/COPYING.txt). Include the license in the `licenses` directory and add a comment in the code giving proper attribution. Ensure any other requirements set forth by the license and/or author are satisfied. **It is your responsibility to disclose code, documentation, or other content derived from external source(s).** If you have questions about whether something can be included in the project or how to give proper attribution, include those questions in your pull request and a reviewer will assist you.

* [x] **This pull request does not include code, documentation, or other content derived from external source(s).**

**Note:** [REVIEWING.md](https://github.com/biocore/scikit-bio/blob/master/REVIEWING.md) may also be helpful to see some of the things code reviewers will be verifying when reviewing your pull request.

### CODE REVIEW

**Comment by mortonjt** [comment]
Thanks @essicolo !  Could you add in a small test case that illustrates when this problem occurs?

**Comment by essicolo** [comment]
Test case.

```
import pandas as pd
import skbio as sb

varespec = pd.read_csv("https://gist.githubusercontent.com/essicolo/cd5c8b77c91e14b9fe648d63f9afaed9/raw/4c44a3d6e39f3d9cfb21aa0ac2d99c13c2997fcb/varespec.csv", delimiter=',', index_col="ID")

varechem = pd.read_csv("https://gist.githubusercontent.com/essicolo/087e49c8feb436f45df5d9e8fa9597f8/raw/d708dea9a107d453dec3073bc7daf70e85870e2c/varechem.csv", delimiter=',', index_col="ID")

rda_test = sb.stats.ordination.rda(x=varechem, y=varespec)
```


```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/home/essicolo/bin/anaconda3/lib/python3.6/site-packages/pandas/core/internals.py in create_block_manager_from_blocks(blocks, axes)
   4293                 blocks = [make_block(values=blocks[0],
-> 4294                                      placement=slice(0, len(axes[0])))]
   4295 

/home/essicolo/bin/anaconda3/lib/python3.6/site-packages/pandas/core/internals.py in make_block(values, placement, klass, ndim, dtype, fastpath)
   2718 
-> 2719     return klass(values, ndim=ndim, fastpath=fastpath, placement=placement)
   2720 

/home/essicolo/bin/anaconda3/lib/python3.6/site-packages/pandas/core/internals.py in __init__(self, values, placement, ndim, fastpath)
    114                              'implies %d' % (len(self.values),
--> 115                                              len(self.mgr_locs)))
    116 

ValueError: Wrong number of items passed 24, placement implies 23

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<ipython-input-1-029b021edb49> in <module>()
      6 varechem = pd.read_csv("https://gist.githubusercontent.com/essicolo/087e49c8feb436f45df5d9e8fa9597f8/raw/d708dea9a107d453dec3073bc7daf70e85870e2c/varechem.csv", delimiter=',', index_col="ID")
      7 
----> 8 rda_test = sb.stats.ordination.rda(x=varechem, y=varespec)

/home/essicolo/bin/anaconda3/lib/python3.6/site-packages/skbio/stats/ordination/_redundancy_analysis.py in rda(y, x, scale_Y, scaling)
    211     biplot_scores = pd.DataFrame(biplot_scores,
    212                                  index=x.columns,
--> 213                                  columns=pc_ids[:biplot_scores.shape[1]])
    214     # The "Correlations of environmental variables with sample
    215     # scores" from table 11.4 are quite similar to vegan's biplot

/home/essicolo/bin/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py in __init__(self, data, index, columns, dtype, copy)
    304             else:
    305                 mgr = self._init_ndarray(data, index, columns, dtype=dtype,
--> 306                                          copy=copy)
    307         elif isinstance(data, (list, types.GeneratorType)):
    308             if isinstance(data, types.GeneratorType):

/home/essicolo/bin/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py in _init_ndarray(self, values, index, columns, dtype, copy)
    481             values = maybe_infer_to_datetimelike(values)
    482 
--> 483         return create_block_manager_from_blocks([values], [columns, index])
    484 
    485     @property

/home/essicolo/bin/anaconda3/lib/python3.6/site-packages/pandas/core/internals.py in create_block_manager_from_blocks(blocks, axes)
   4301         blocks = [getattr(b, 'values', b) for b in blocks]
   4302         tot_items = sum(b.shape[0] for b in blocks)
-> 4303         construction_error(tot_items, blocks[0].shape[1:], axes, e)
   4304 
   4305 

/home/essicolo/bin/anaconda3/lib/python3.6/site-packages/pandas/core/internals.py in construction_error(tot_items, block_shape, axes, e)
   4278         raise ValueError("Empty data passed with indices specified.")
   4279     raise ValueError("Shape of passed values is {0}, indices imply {1}".format(
-> 4280         passed, implied))
   4281 
   4282 

ValueError: Shape of passed values is (24, 14), indices imply (23, 14)
```

**Comment by mortonjt** [comment]
Perfect.  Can we upload those datasets along with a unittest in [test_redundancy_analysis.py](https://github.com/biocore/scikit-bio/blob/master/skbio/stats/ordination/tests/test_redundancy_analysis.py)?

**Comment by essicolo** [comment]
I'm not sure if it's legit since data were exported from the vegan R package, licensed GPL2. On the other hand, these are published data which can be copy-pasted from a published paper.

**Comment by mortonjt** [comment]
If it is published - it should be fine.

**Comment by essicolo** [comment]
I'm not sure I've done that by the book (probably not). Eigen values and proportion explained returned different results comparing to vegan. I have just changed the pandas indexes, so it's probably inherited from the master branch.

**Comment by essicolo** [comment]
That's an other subject, but for its eigenvalues output (`summary(rda_, scaling=1)$cont$importance`) vegan outputs the eigen values of the predicted matrix, i.e. `eigen(cov(Y_hat)$values)`, whereas the `rda` function outputs the singular values `s` in the scipy `svd` function. I shouldn't expected them to be equal.

File: skbio/stats/ordination/tests/test_redundancy_analysis.py [hunk 1]
```diff
@@ -165,6 +165,64 @@ def test_scaling2(self):
...
+class TestRDAResults_biplot_score(TestCase):
+    # STATUS: L&L only shows results with scaling 1, and they agree
+    # with vegan's (module multiplying by a constant). I can also
+    # compute scaling 2, agreeing with vegan, but there are no written
+    # results in L&L.
+    def setUp(self):
+        """varespec and varechem from Väre etal. 1995 DOI: 10.2307/3236351"""
+
+        self.Y = read_csv(get_data_path('varespec.csv'))
+        self.X = read_csv(get_data_path('varechem.csv'))
```
Comment (mortonjt) [inline skbio/stats/ordination/tests/test_redundancy_analysis.py]: I don't see read_csv imported, so these lines will need to be corrected to
```
self.Y = pd.read_csv(get_data_path('varespec.csv'))
self.X = pd.read_csv(get_data_path('varechem.csv'))
```

File: skbio/stats/ordination/tests/test_redundancy_analysis.py [hunk 2]
```diff
@@ -165,6 +165,64 @@ def test_scaling2(self):
...
+
+    def test_biplot_score(self):
+
+        rda_ = rda(y=varespec, x=varechem, scale_Y=False, scaling=1)
```
Comment (mortonjt) [inline skbio/stats/ordination/tests/test_redundancy_analysis.py]: This needs to match up with the variables in `setUp`

File: skbio/stats/ordination/tests/test_redundancy_analysis.py [hunk 3]
```diff
@@ -165,6 +165,64 @@ def test_scaling2(self):
...
+            samples=vegan_samples,
+            features=vegan_features,
+            sample_constraints=vegan_constraints,
+            biplot_scores=vegan_biplot,
+            proportion_explained=vegan_propexpl,
+            eigvals=vegan_eigvals)
+
+        assert_ordination_results_equal(rda_, exp,
+                                        ignore_directionality=True,
+                                        decimal=6)
```
Comment (mortonjt) [inline skbio/stats/ordination/tests/test_redundancy_analysis.py]: If the above two comments are corrected, the results here don't seem to agree ...

Comment (essicolo) [reply skbio/stats/ordination/tests/test_redundancy_analysis.py]: biplot_scores are still different from vegan (only the first score column is equal). Any help to figure out why would be appreciated.

**Comment by essicolo** [comment]
The failing check is about the difference between bitplot scores from this code and from vegan. The difference is a scaling factor. I reviewed the math and the code from skbio fits with Legendre and Legendre. I couldn't figure out how vegan is generating its biplot scores.

**Comment by mortonjt** [comment]
Turns out this is because that vegan scales their biplot scores by a factor of eig[0] / eig[:]

Opening up another PR to fix this

**Comment by RNAer** [comment]
close in favor of #1557

**Pull request closed without merging**